x/httpproxy/connect_handler.go

```diff
@@ -25,47 +24,83 @@ import (
 
 type handler struct {
 	dialer transport.StreamDialer
+	client http.Client
 }
 
 var _ http.Handler = (*handler)(nil)
 
 // ServeHTTP implements [http.Handler].ServeHTTP for CONNECT requests, using the internal [transport.StreamDialer].
-func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (h *handler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.Request) {
 	// TODO(fortuna): For public services (not local), we need authentication and drain on failures to avoid fingerprinting.
-	if r.Method != http.MethodConnect {
-		http.Error(w, fmt.Sprintf("Method %v is not supported", r.Method), http.StatusMethodNotAllowed)
+	if proxyReq.Method == http.MethodConnect {
+		h.handleConnect(proxyResp, proxyReq)
 		return
+	} else if proxyReq.URL.Host != "" {
+		h.handleHTTPProxyRequest(proxyResp, proxyReq)
+	} else {
+		http.Error(proxyResp, "Not Found", http.StatusNotFound)
 	}
+}
+
+func (h *handler) handleHTTPProxyRequest(proxyResp http.ResponseWriter, proxyReq *http.Request) {
+	targetReq, err := http.NewRequestWithContext(proxyReq.Context(), proxyReq.Method, proxyReq.URL.String(), proxyReq.Body)
```

Comment:
Are there any specific reasons why we don't reuse `proxyReq`, but instead copy it to a new `targetReq`?

Reply:
Yes, the proxyReq uses an absolute URL, while the targetReq uses relative path + Host header.

I added a comment.